floodlight/core/events.py

```diff
@@ -61,3 +74,155 @@ def custom(self):
         )
 
         return custom
+
+    def check(self):
+        """Checks an Event object and throws an error if any mandatory column misses
+
+        It is checked if the Event object contains the mandatory columns (defined in
+        floodlight.core.definitions) and an error is thrown if this check fails.
+        Additionally, all essential and all protected columns from the Event object are
+        checked for dtypes and value ranges from the definitions, however, if these
+        checks fail there is only a warning at runtime.
+        """
+        # check for essential columns and throw error if check fails
+        if not self.check_for_essential_cols():
+            raise TypeError(
+                "Found irregular Events object missing at least one essential column!"
+            )
+
+        # check dtypes and value range of essential and protected columns
+        self.check_essential_cols()
+        self.check_protected_cols()
+
+    def check_for_essential_cols(self) -> bool:
+        """Checks if the mandatory essential columns exist in the inner events DataFrame
+
+        Returns
+        -------
+        check: bool
+            True if all essential columns are in self.events and False otherwise
+        """
+        check = True
+        for col in essential_events_columns:
+            if col not in self.essential:
+                check = False
+                warnings.warn(
```

Comment:
This seems to be superfluous to me. When you call this method in `check()` with a missing column, it seems as this will throw a warning first and then the error right afterwards. I think the purpose of this method is well achieved with returning a bool.

I agree though that knowing the missing column name would be nice. Maybe refactor to a method (or even property) of the like "missing_essential_columns" that returns a list of missing essential columns, or None if everything is fine. Then, in `check()`, you can go like
```
If self.missing_essential_columns() is not None:
   raise ValueError...